src/extension/ncdfgenerator/src/main/java/au/org/emii/ncdfgenerator/FeatureTypeFactory.java

```diff
@@ -0,0 +1,110 @@
+package au.org.emii.ncdfgenerator;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import org.geotools.jdbc.VirtualTable;
+import org.geotools.data.Query;
+import org.geotools.data.Transaction;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.jdbc.JDBCDataStore;
+import org.opengis.feature.type.*;
+import org.geotools.data.simple.SimpleFeatureIterator;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+
+import com.vividsolutions.jts.geom.Geometry;
+
+public class FeatureTypeFactory {
+    // Lifted from: https://github.com/geoserver/geoserver/blob/master/src/web/core/src/main/java/org/geoserver/web/data/layer/SQLViewAbstractPage.java#L404
```

Comment:
Ah ok I see.  Yeah I had thought that too actually but I was too lazy to change what I had working!  But I will, because it's a good idea!